### DISCUSSION TITLE
Dynamic API routes cannot be called in 'getServerSideProps'

### DISCUSSION THREAD

**Discussion by abidk** [vercel/next.js#11921] [Help]
It's not possible to calling dynamic api routes in 'getServerSideProps'. A 404 error is returned instead.

Steps to reproduce:
1. Create an /api/products/data.js etc.
2. Call endpoint 'http://locahost:3000/api/products/data' in getServerSideProps.
3. 404 returned

Calling the api endpoint in a browser works as expected.

**Comment by paales** [comment]
`getServerSideProps` already runs on the server, maybe you can call that code directly?

**Reply by timneutkens** [reply]
Yep @paales is right, there's no reason to introduce additional http overhead because you can import the underlying function directly and execute it.
